### PR TITLE
[#2209] [FOLLOWUP]feat(server) Support config buffer size for HadoopShuffleWriteHandler

### PR DESCRIPTION
**Pull request by lwllvyb**
### What changes were proposed in this pull request?

Support config buffer size for HadoopShuffleWriteHandler

### Why are the changes needed?

Fix: #2209 

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Locally test.

### CODE REVIEW

**Review by jerqi — COMMENTED** [review]
Should this config option be same with the local? Maybe they should have different default value for better performance.

**Comment by lwllvyb** [comment]
> Should this config option be same with the local? Maybe they should have different default value for better performance.

Thanks for your suggestion. I am very agree with you that they will get better performance with different values for different storages.
I will add different configs (```rss.storage.localfile.write.dataBufferSize```, ```rss.storage.hadoop.write.indexBufferSize```, ```rss.storage.hadoop.write.dataBufferSize```, ```rss.storage.hadoop.write.indexBufferSize```) for LOCALFILE and HADOOP. 
If we can set the default values to be the same (e.g. 8K ?), we can config different values to cope with different real cluster scenarios. 
Do you have any better suggestions on default values?

Thanks again for taking the time to review.

**Comment by jerqi** [comment]
> > Should this config option be same with the local? Maybe they should have different default value for better performance.
> 
> Thanks for your suggestion. I am very agree with you that they will get better performance with different values for different storages. I will add different configs (`rss.storage.localfile.write.dataBufferSize`, `rss.storage.hadoop.write.indexBufferSize`, `rss.storage.hadoop.write.dataBufferSize`, `rss.storage.hadoop.write.indexBufferSize`) for LOCALFILE and HADOOP.
> 
> If we can set the default values to be the same (e.g. 8K ?), we can config different values to cope with different real cluster scenarios.
> 
> Do you have any better suggestions on default values? Thanks again for taking the time to review.

Ok for me.

**Comment by lwllvyb** [comment]
> > Should this config option be same with the local? Maybe they should have different default value for better performance.
> 
> Thanks for your suggestion. I am very agree with you that they will get better performance with different values for different storages. I will add different configs (`rss.storage.localfile.write.dataBufferSize`, `rss.storage.hadoop.write.indexBufferSize`, `rss.storage.hadoop.write.dataBufferSize`, `rss.storage.hadoop.write.indexBufferSize`) for LOCALFILE and HADOOP.
> 
> If we can set the default values to be the same (e.g. 8K ?), we can config different values to cope with different real cluster scenarios.
> 
> Do you have any better suggestions on default values? Thanks again for taking the time to review.

To be consistent with ```rss.storage.type```'s values, I update the configs from `rss.storage.hadoop.write.dataBufferSize`, `rss.storage.hadoop.write.indexBufferSize` to `rss.storage.hdfs.write.dataBufferSize`, `rss.storage.hdfs.write.indexBufferSize`

File: common/src/main/java/org/apache/uniffle/common/config/RssBaseConf.java [hunk 1]
```diff
@@ -294,17 +294,29 @@ public class RssBaseConf extends RssConf {
...
+      ConfigOptions.key("rss.storage.localfile.write.indexBufferSize")
           .stringType()
           .defaultValue("8k")
-          .withDescription("The buffer size to cache the write index content.");
+          .withDescription("The buffer size to cache the write index content for LOCALFILE.");
+
+  public static final ConfigOption<String> RSS_STORAGE_HDFS_WRITE_DATA_BUFFER_SIZE =
+      ConfigOptions.key("rss.storage.hdfs.write.dataBufferSize")
+          .stringType()
+          .defaultValue("8k")
```
Comment (jerqi) [inline common/src/main/java/org/apache/uniffle/common/config/RssBaseConf.java:312]: Is `8k` ok for HDFS? Maybe we would better use Hadoop default config option before we have a benchmark.

Comment (jerqi) [reply common/src/main/java/org/apache/uniffle/common/config/RssBaseConf.java:312]: OK.

**Review by jerqi — APPROVED** [review]
LGTM.

**Pull request merged**